plugins/src/main/java/opengrok/auth/plugin/ldap/LdapServer.java

```diff
@@ -122,13 +129,89 @@ public void setInterval(int interval) {
         this.interval = interval;
     }
 
+    private String urlToHostname(String urlStr) throws URISyntaxException {
+        URI uri = new URI(urlStr);
+        return uri.getHost();
+    }
+
+    /**
+     * This method converts the scheme from URI to port number.
+     * It is limited to the ldap/ldaps schemes.
+     * The method could be static however then it cannot be easily mocked in testing.
+     * @return port number
+     * @throws URISyntaxException if the URI is not valid
+     */
+    public int getPort() throws URISyntaxException {
+        URI uri = new URI(getUrl());
+        switch (uri.getScheme()) {
+            case "ldaps":
+                return 636;
+            case "ldap":
+                return 389;
+        }
+
+        return -1;
+    }
+
+    private boolean isReachable(InetAddress addr, int port, int timeOutMillis) {
+        try {
+            try (Socket soc = new Socket()) {
+                soc.connect(new InetSocketAddress(addr, port), timeOutMillis);
```

Comment:
`InetAddress.isReachable()` is not really useful as it either uses ICMP echo or TCP echo service. The former is not usable out of the box on Linux due to limited privileges, the latter is useless in this case as we are interested in other ports (and the TCP echo service is very little used these days anyway, if at all).